pydantic/experimental/pipeline.py

```diff
@@ -434,6 +437,16 @@ def _apply_transform(s: cs.CoreSchema | None, func: Callable[[Any], Any]) -> cs.
             s = s.copy()
             s['to_upper'] = True
             return s
+
+    # TODO: need to expand this logic to cover more cases, like datetimes, etc
```

Comment:
Is this to support `transform(str)`? If so why not tell users to use `validate_as(str)`?

Reply:
Yeah we can... I was trying to improve JSON schema for this case @dmontagu mentioned:

```
ta = TypeAdapter(Annotated[int, validate_as(str).transform(int)])
```

Reply:
Removing 😎